app-modules/prospect/src/DataTransferObjects/UpdateProspectData.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2025, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Prospect\DataTransferObjects;
+
+use Spatie\LaravelData\Attributes\MapName;
+use Spatie\LaravelData\Data;
+use Spatie\LaravelData\Mappers\SnakeCaseMapper;
+use Spatie\LaravelData\Optional;
+
+#[MapName(SnakeCaseMapper::class)]
+class UpdateProspectData extends Data
+{
+    public function __construct(
+        public string | Optional | null $firstName,
+        public string | Optional | null $lastName,
+        public string | Optional | null $fullName,
+        public string | Optional | null $preferred,
+        public string | Optional | null $description,
+        public bool | Optional | null $smsOptOut,
+        public bool | Optional | null $emailBounce,
+        public string | null $status,
+        public string | null $source,
```

Comment:
I am pretty sure we do not allow `status` or `source` on a Prospect to be `null` from a database perspective or in your controller. So I think `null` should be removed here.

I think you may have been thinking of `Optional` which his appropriate here in the update as they could just not be wanting to update this.